hunting/aws/docs/ec2_modify_instance_attribute_user_data.md

```diff
@@ -0,0 +1,42 @@
+# EC2 Modify Instance Attribute User Data
+
+---
+
+## Metadata
+
+- **Author:** Elastic
+- **Description:** This hunting query identifies when a user modifies the user data attribute of an EC2 instance. The user data attribute is a script that runs when the instance is launched. Modifying the user data attribute could indicate an adversary attempting to gain persistence or execute malicious code on the instance.
+
+- **UUID:** `f11ac62c-5f42-11ef-9d72-f661ea17fbce`
+- **Integration:** [aws.cloudtrail](https://docs.elastic.co/integrations/aws/cloudtrail)
+- **Language:** `[ES|QL]`
+- **Source File:** [EC2 Modify Instance Attribute User Data](../queries/ec2_modify_instance_attribute_user_data.toml)
+
+## Query
+
+```sql
+from logs-aws.cloudtrail-*
+| where @timestamp > now() - 7 day
+| where
+    event.provider == "ec2.amazonaws.com"
+    and event.action == "ModifyInstanceAttribute"
+    and aws.cloudtrail.request_parameters RLIKE ".*attribute=userData.*"
+| dissect aws.cloudtrail.request_parameters "{%{instance_id_key}=%{instance_id}, %{attribute_key}=%{attribute}, %{value_key}=%{value}}"
+| stats user_data_upload_counts = count(*) by aws.cloudtrail.user_identity.arn, event.outcome
```

Comment:
```suggestion
| stats user_data_upload_counts = count(*) by aws.cloudtrail.user_identity.arn, event.outcome
```

user_data_upload_counts or user_attribute_modify_count ?

Reply:
Changed to `user_attribute_modify_count`